### PR TITLE
Fix tests and implement MakerOracle.

### PR DESCRIPTION
**Pull request by Kenadia**
@BrendanChou wanted to let you take a look over this.

Some todos, could be in this PR or other:
- [x] in tests, use a third-party address to make the margin calls
- [x] test cases for unsupported tokens
- [ ] consider doing the math differently if overflow is an issue

### CODE REVIEW

**Comment by BrendanChou** [comment]
1. Consider combining `positionUsesSupportedTokens()` into `meetsCollateralRequirement()`
2. Use SafeMath
3. Consider using constants for (10**8) and (10**18)

File: contracts/margin/external/oracle/MakerOracle.sol [hunk 1]
```diff
@@ -167,35 +144,36 @@ contract MakerOracle is
...
             return (
-                heldAmount * (10**8) * (10**18) >=
-                owedAmount * COLLATERAL_REQUIREMENT * ethUsdRate
-            );
-        } else if (heldToken == WETH && owedToken == DAI) {
-            // I.e. long position in ETH.
-            return (
-                heldAmount * (10**8) * ethUsdRate >=
-                owedAmount * COLLATERAL_REQUIREMENT * (10**18)
+                heldAmount.mul(10**26) >=
```
Comment (BrendanChou) [inline contracts/margin/external/oracle/MakerOracle.sol]: It's a little more unclear with  the `10**26` and the `10**10`. Can you put in a comment explaining what you're doing here, or change it to `.mul((10**8) * (10**18))`

Comment (Kenadia) [reply contracts/margin/external/oracle/MakerOracle.sol]: Added comments

**Comment by Kenadia** [comment]
PTAL

**Pull request merged**